### PR TITLE
LTC 7 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/the_black_gate.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Name:The Black Gate
+ManaCost:no cost
+Types:Legendary Land Gate
+K:ETBReplacement:Other:DBTap
+SVar:DBTap:DB$ Tap | ETB$ True | Defined$ Self | UnlessCost$ PayLife<3> | UnlessPayer$ You | StackDescription$ enters the battlefield tapped. | SpellDescription$ As CARDNAME enters the battlefield, you may pay 3 life. If you don't, it enters the battlefield tapped.
+A:AB$ Mana | Cost$ T | Produced$ B | SpellDescription$ Add {B}.
+A:AB$ ChoosePlayer | Cost$ 1 B T | Defined$ You | Choices$ Player.withMostLife | ChoiceTitle$ Choose a player with the most life or tied for most life | SubAbility$ DBEffect | SpellDescription$ Choose a player with the most life or tied for most life. Target creature can't be blocked by creatures that player controls this turn.
+SVar:DBEffect:DB$ Effect | ValidTgts$ Creature | RememberObjects$ ChosenPlayer,Targeted | StaticAbilities$ CantBlock | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_black_gate.txt:8]: needs cleanup

File: forge-gui/res/cardsfolder/upcoming/field_tested_frying_pan.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+SVar:TrigToken:DB$ Token | TokenScript$ c_a_food_sac | SubAbility$ DBToken
+SVar:DBToken:DB$ Token | TokenScript$ w_1_1_halfling | RememberTokens$ True | SubAbility$ DBAttach
+SVar:DBAttach:DB$ Attach | Defined$ Remembered | Object$ Self | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+S:Mode$ Continuous | Affected$ Creature.EquippedBy | AddTrigger$ GainLifeTrig | Description$ Equipped creature has "Whenever you gain life, this creature gets +X/+X until end of turn, where X is the amount of life you gained."
+SVar:GainLifeTrig:Mode$ LifeGained | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigPump | TriggerDescription$ Whenever you gain life, this creature gets +X/+X until end of turn, where X is the amount of life you gained.
+SVar:TrigPump:DB$ Pump | Defined$ Self | NumAtt$ +X | NumDef$ +X
+SVar:X:TriggerCount$LifeAmount
+DeckHas:Ability$Tokens|LifeGain|Sacrifice & Type$Food|Artifact|Halfling
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/field_tested_frying_pan.txt:14]: invalid Ability

File: forge-gui/res/cardsfolder/upcoming/gilraen_dunedain_protector.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+SVar:DBReturn:DB$ ChangeZone | Defined$ Targeted | Origin$ All | Optional$ True | Destination$ Battlefield | RememberChanged$ True | SubAbility$ DBDelayedTrigger
+SVar:DBDelayedTrigger:DB$ DelayedTrigger | ConditionDefined$ Remembered | ConditionPresent$ Card | ConditionCompare$ EQ0 | RememberObjects$ RememberedLKI | Mode$ Phase | Phase$ End Of Turn | Execute$ TrigReturn | SubAbility$ DBCleanup | SpellDescription$  If you don't, at the beginning of the next end step, return that card to the battlefield under its owner's control with a vigilance counter and a lifelink counter on it.
+SVar:TrigReturn:DB$ ChangeZone |  Defined$ DelayTriggerRememberedLKI | Origin$ Exile | Destination$ Battlefield | WithCountersType$ Vigilance,Lifelink | WithCountersAmount$ 1
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/gilraen_dunedain_protector.txt:8]: card untested imo

File: forge-gui/res/cardsfolder/upcoming/monstrosity_of_the_lake.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Name:Monstrosity of the Lake
+ManaCost:4 U
+Types:Legendary Creature Kraken
+PT:4/6
+T:Mode$ ChangesZone | ValidCard$ Card.Self | Destination$ Battlefield | Execute$ TrigTapAll | TriggerDescription$ When When CARDNAME enters the battlefield, you may pay {5}. If you do, tap all creatures your opponents control, then put a stun counter on each of those creatures. (If a permanent with a stun counter would become untapped, remove one from it instead.)
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/monstrosity_of_the_lake.txt:5]: When When

File: forge-gui/res/cardsfolder/upcoming/moria_scavenger.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
+Name:Moria Scavenger
+ManaCost:1 B R
+Types:Creature Orc Rogue
+PT:1/4
+K:Deathtouch
+K:Haste
+A:AB$ Draw | Cost$ T Discard<1/Card> | NumCards$ 1 | SubAbility$ DBAmass | SpellDescription$ Draw a card. If the discarded card was a creature card, amass Orcs 1. (Put a +1/+1 counter on an Army you control. It's also an Orc. If you don't control an Army, create a 0/0 black Orc Army creature token first.)
+SVar:DBAmass:DB$ Amass | Type$ Orc | Num$ 1 | ConditionCheckSVar$ X | ConditionSVarCompare$ GE1
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/moria_scavenger.txt:8]: without extra SVar please

File: forge-gui/res/cardsfolder/upcoming/the_black_gate.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_black_gate.txt:8]: wrong cleanup

File: forge-gui/res/cardsfolder/upcoming/gilraen_dunedain_protector.txt (hunk 3 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/gilraen_dunedain_protector.txt]: Yeah forgot to note that the delayedtrigger isn't working. Any pointers to cards I should look at ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/gilraen_dunedain_protector.txt]: What doesn't work now?
(Except for the missing counters)

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/gilraen_dunedain_protector.txt]: it triggers at end of turn (if you chose to not return it immediately) but nothing gets returned

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/gilraen_dunedain_protector.txt]: you might want to draw a diagram of the effect chain for yourself and in what cases something would get remembered to understand what is happening

File: forge-gui/res/cardsfolder/upcoming/the_black_gate.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/the_black_gate.txt:8]: ClearChosenPlayer is not correct ?

File: forge-gui/res/cardsfolder/upcoming/gilraen_dunedain_protector.txt (hunk 3 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/gilraen_dunedain_protector.txt]: got the delayedtrigger to work though not the etb counters. Does this need an additional VAR ?

File: forge-gui/res/cardsfolder/upcoming/the_black_gate.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/the_black_gate.txt:8]: You mean some sort of hidden static trigger ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_black_gate.txt:8]: Don't do cleanup for something that's not needed

**Review by tool4ever — APPROVED** [review]

**Pull request merged**